core/templates/dev/head/pages/exploration_editor/translation_tab/TranslationRecordingService.ts

```diff
@@ -0,0 +1,169 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the 'License');
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an 'AS-IS' BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for handling microphone data and mp3 audio processing.
+ */
+oppia.factory('TranslationRecordingService', ['$q', '$window',
+  function($q, $window) {
+    var AudioContext = null,
+      defer = null,
+      definedAudioContext = null, // will be defined audio context
+      isAvailable = null,
+      isRecording = false,
+      microphone = null,
+      microphoneStream = null,
+      mp3Worker = null,
+      processor = null;
+
+
+    var _initWorker = function() {
+      if (!$window.Worker) {
+        console.warn('Worker API not supported in this browser.');
```

Comment:
It cannot send the mic data to worker for converting. I'll have boolean based check to prevent start recording and make it available to the ui via status method.

Reply:
Changed to check if worker is available instead via null check.

Reply:
Done.